src/value/owned.rs

```diff
@@ -120,12 +168,25 @@ impl Struct for OwnedStruct {
 
     fn iter<'a>(
         &'a self,
-    ) -> Box<dyn Iterator<Item = (&'a Self::FieldName, &'a Self::Element)> + 'a> {
-        // convert &(k, v) -> (&k, &v)
-        Box::new(self.fields.iter().map(|kv| match &kv {
-            (k, v) => (k, v),
+    ) -> Box<dyn Iterator<Item = (&'a Self::FieldName, Option<&'a Self::Element>)> + 'a> {
+        // convert (&k, &v[0..n]) -> (&k, &v[0])
+        Box::new(self.fields.iter().map(|kv| match kv {
+            (k, v) => (k, v.first()),
         }))
     }
+
+    fn get(&self, field_name: &Self::FieldName) -> Option<&Self::Element> {
+        self.fields.get(field_name)?.first()
+    }
+
+    // TODO remove usage of unwrap()[unwrap] as it will panic for None
+    //  [unwrap] https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap
+    fn get_all<'a>(
+        &'a self,
+        field_name: &'a Self::FieldName,
+    ) -> Box<dyn Iterator<Item = &'a Self::Element> + 'a> {
+        Box::new(self.fields.get(field_name).unwrap().iter())
+    }
```

Comment:
One thing to consider is that `Option<T>` is a container so you can iterate/map over it.  Consider this [snippet][1]:

```rust
fn mk_iter<'a, T>(opt: Option<&'a Vec<T>>) -> Box<dyn Iterator<Item = &'a T> + 'a> {
    Box::new(opt.into_iter().flat_map(|v| v.iter()))
}

fn to_string_vec<'a, I: Iterator<Item = &'a String> + 'a>(iter: I) -> Vec<String> {
    iter.map(|e| e.to_owned()).collect()
}

fn main() {
    let vals = vec!["a", "b", "c"].iter().map(|s| s.to_string()).collect();
    let some_vals = Some(&vals);
    let new_some_vals: Vec<String> = to_string_vec(mk_iter(some_vals));
    assert_eq!(vals, new_some_vals);

    let none_vals: Option<&Vec<String>> = None;
    let new_none_vals: Vec<String> = to_string_vec(mk_iter(none_vals));
    assert!(new_none_vals.is_empty());
}
```

[1]: https://play.rust-lang.org/?version=stable&edition=2018&gist=f64e8eadf682f82480a86da03549c4f8

Reply:
Thanks! This works fine for `get_all`.